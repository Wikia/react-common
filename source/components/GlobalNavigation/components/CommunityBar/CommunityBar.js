import React from 'react';
import PropTypes from 'prop-types';

import LogoFandomHeart from '../../../../assets/LogoFandomHeart';

const CommunityBar = ({ model, siteName }) => {
    const { href, title, tracking_label: trackingLabel, mobile_title: mobileTitle } = siteName;

    return (
        <div className="wds-global-navigation__community-bar">
            <a
                href={model.href}
                className="wds-global-navigation__logo-heart-link"
                data-tracking-label={model['tracking-label']}
            >
                <LogoFandomHeart className="wds-global-navigation__logo-heart" />
            </a>
            <div className="wds-global-navigation__community-bar-separator" />
            <div className="wds-community-bar__sitename">
                <a
                    href={href}
                    data-tracking-label={trackingLabel}
                >
                    {
                        mobileTitle ? mobileTitle.value : title.value
                    }
                </a>
            </div>
        </div>
    );
};

CommunityBar.propTypes = {
    model: PropTypes.shape().isRequired,
    siteName: PropTypes.shape().isRequired,
};

export default CommunityBar;
