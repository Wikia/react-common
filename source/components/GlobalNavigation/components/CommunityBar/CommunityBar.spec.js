import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import CommunityBar from './CommunityBar';

const defaultProps = {
    siteName: {
        title: { type: 'text', value: 'Xkxd Wiki' },
        mobile_title: { type: 'text', value: 'Xkxd Wiki' },
        href: '//xkxd.fandom.com',
        tracking_label: 'sitename',
    },
    model: {
        'tracking-label': 'track-logo',
    },
};

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<CommunityBar {...computedProps} />);
}

test('CommunityBar renders correctly with default props', () => {
    expect(renderComponent()).toMatchSnapshot();
});

test('CommunityBar renders correctly when mobileTitle is empty', () => {
    expect(renderComponent({ siteName: { mobile_title: null } })).toMatchSnapshot();
});
