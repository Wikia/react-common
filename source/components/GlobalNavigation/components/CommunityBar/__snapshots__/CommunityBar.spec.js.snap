// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityBar renders correctly when mobileTitle is empty 1`] = `
<div
  className="wds-global-navigation__community-bar"
>
  <a
    className="wds-global-navigation__logo-heart-link"
    data-tracking-label="track-logo"
  >
    <_default
      className="wds-global-navigation__logo-heart"
    />
  </a>
  <div
    className="wds-global-navigation__community-bar-separator"
  />
  <div
    className="wds-community-bar__sitename"
  >
    <a
      data-tracking-label="sitename"
      href="//xkxd.fandom.com"
    >
      Xkxd Wiki
    </a>
  </div>
</div>
`;

exports[`CommunityBar renders correctly with default props 1`] = `
<div
  className="wds-global-navigation__community-bar"
>
  <a
    className="wds-global-navigation__logo-heart-link"
    data-tracking-label="track-logo"
  >
    <_default
      className="wds-global-navigation__logo-heart"
    />
  </a>
  <div
    className="wds-global-navigation__community-bar-separator"
  />
  <div
    className="wds-community-bar__sitename"
  >
    <a
      data-tracking-label="sitename"
      href="//xkxd.fandom.com"
    >
      Xkxd Wiki
    </a>
  </div>
</div>
`;
